client_src/js/lib/modules/printerSettings.js

```diff
@@ -1,1325 +1,1403 @@
-import OctoFarmClient from "../octofarm.js";
+/* eslint-disable quotes */
+import OctoFarmClient from "../octofarm_client.js";
 import UI from "../functions/ui.js";
-import Script from "./scriptCheck.js";
 import Calc from "../functions/calc.js";
+import Script from "./scriptCheck.js";
 
-let currentIndex = 0;
-
-// let controlDropDown = false;
-
-let currentPrinter = null;
+let currentPrinterIndex;
+let printerOnline;
+let currentPrinter;
+let hasErrorNotificationBeenTriggered = false;
+let pageElements;
 
 // Close modal event listeners...
-$("#PrinterSettingsModal").on("hidden.bs.modal", function (e) {
-  // Fix for mjpeg stream not ending when element removed...
-  document.getElementById("printerControlCamera").src = "";
-  // Remove buggy drop down
-  // if (document.getElementById("printerSelection")) {
-  //   document.getElementById("printerSelection").remove();
-  //   controlDropDown = false;
-  // }
-});
 $("#connectionModal").on("hidden.bs.modal", function (e) {
   if (document.getElementById("connectionAction")) {
     document.getElementById("connectionAction").remove();
   }
 });
 
-export default class PrinterSettings {
-  static async init(index, printers, printerControlList) {
-    try {
-      if (index !== "") {
-        UI.clearSelect("ps");
-        const printerProfileBtn = document.getElementById(
-          "printer-profile-btn"
+async function updatePrinterSettingsModal(printersInformation, printerID) {
```

Comment:
This function is fired initially from the API call. It receives the printer ID and updates the rest of the page with that specific printer. 
It's then fired again from the SSE event, without a printer ID this time. The variables above will be filled by this time with the specific printers ID so we can continue updating the live elements. It's mainly the printer status stuff at the bottom of the page.